File: app/models/fastui/m_tab.rb
# encoding: utf-8
module Fastui
  class MTab < MObject
    attr_accessible :m_window, :m_window_id, :m_entity, :m_entity_id, :ro
    attr_accessible :included_tab, :included_tab_id

    belongs_to :m_window, :class_name => 'Fastui::MWindow'
    belongs_to :m_entity, :class_name => 'Fastui::MEntity'
    belongs_to :included_tab, :class_name => 'Fastui::MTab'
    has_many :m_fields, :dependent => :destroy
    has_many :m_columns, :dependent => :destroy


  end
end

File: app/assets/javascripts/fastui/app/model/MTab.js
Ext.define('FastUI.model.MTab', {
    extend:'Ext.data.Model',
    requires:['FastUI.model.MField','FastUI.model.MColumn'],
    fields:['id', 'title', 'm_window_id','model_class','note','help','actived','org_id','createdby','updatedby'],
    associations:[
        {
            type:'belongsTo',
            model:'FastUI.model.MWindow',
            name:'mwindow'
//            primaryKey:'id',
//            foreignKey:'m_window_id'
            //associationKey: 'parent_group' // read parent data from parent_group
        },
        {
            type: 'hasMany',
            model: 'FastUI.model.MField',
            name: 'mfields',
            associationKey: 'm_fields'
        },
        {
            type: 'hasMany',
            model: 'FastUI.model.MColumn',
            name: 'mcolumns',
            associationKey: 'm_columns'
        }
    ],

    proxy:{
        type:'rest',
        url:'/fastui/m_windows/:id/m_tabs',
        format:'json'
//        reader:{
//            type:'json',
//            root:'m_tabs',
//            successProperty:'success'
//        }
    }

});